go/ql/lib/semmle/go/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -227,7 +223,14 @@ class CastNode extends ExprNode {
 
 class DataFlowExpr = Expr;
 
-class DataFlowType = Type;
+private newtype TDataFlowType =
+  TTodoDataFlowType() or
+  TTodoDataFlowType2() // Add a dummy value to prevent bad functionality-induced joins arising from a type of size 1.
```

Comment:
I note only some other languages do this -- might as well avoid the hack if not necessary

Reply:
Those languages did need it though. I assume it wouldn't be hard to remove our hack when the other hacks are removed, if it isn't needed any more. Are you suggesting that I omit the second branch, run a performance check to see if we can get by without it, and if so then omit it and hope that we don't hit the performance problem in future?

Reply:
Basically yes, though I wouldn't object to the flow merge as-is (after the requested double-check below), then experimentally remove the hack from all languages.

Suggest also querying on slack whether the comment about singleton types still applies or has perhaps been fixed since.